src/clp_ffi_js/ir/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,233 @@
+#include "ClpIrV1Decoder.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <span>
+#include <string>
+#include <system_error>
+#include <utility>
+
+#include <clp/ErrorCode.hpp>
+#include <clp/ffi/encoding_methods.hpp>
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/TraceableException.hpp>
+#include <emscripten/em_asm.h>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/ClpJsException.hpp>
+#include <clp_ffi_js/constants.hpp>
+
+using namespace std::literals::string_literals;
+
+namespace clp_ffi_js::ir {
+auto ClpIrV1Decoder::create(emscripten::val const& data_array) -> ClpIrV1Decoder {
+    auto const length{data_array["length"].as<size_t>()};
+    SPDLOG_INFO("ClpIrV1Decoder::ClpIrV1Decoder() got buffer of length={}", length);
+
+    auto data_buffer{std::make_unique<char const[]>(length)};
```

Comment:
clang-tidy suggests using `std::array`, but I think the modern C++ approach is to use a `std::vector` here.

Reply:
We initially had an `std::vector` for this. @LinZhihao-723 and I were discussing this offline and we concluded that a unique pointer to an array like this might be sufficient, considering how [non-straightforward is it to deallocate a vector](https://stackoverflow.com/questions/3054567/right-way-to-deallocate-an-stdvector-object).

I also wasn't too sure how `std::array` would work for this, since I thought `std::array` has a compile-time fixed size?

Reply:
Ah, you can still use a unique pointer with a vector, right? The goal as suggested by clang-tidy is just to avoid using C-style arrays.

Yeah, `std::array` won't work. I think clang-tidy's recommendation is just hard-coded to that.